flink-yarn/src/main/java/org/apache/flink/yarn/YarnClusterDescriptor.java

```diff
@@ -305,23 +305,25 @@ public void addShipFiles(List<File> shipFiles) {
     private void addShipArchives(List<File> shipArchives) {
         checkArgument(
                 isArchiveOnlyIncludedInShipArchiveFiles(shipArchives),
-                "Non-archive files are included.");
+                "Not all files included in shipArchives are archive files.");
         this.shipArchives.addAll(shipArchives);
     }
 
     private static boolean isArchiveOnlyIncludedInShipArchiveFiles(List<File> shipFiles) {
-        return shipFiles.stream()
+        List<String> shipFileNames = shipFiles.stream()
                 .filter(File::isFile)
                 .map(File::getName)
                 .map(String::toLowerCase)
-                .allMatch(
+                .filter(
                         name ->
                                 name.endsWith(".tar.gz")
                                         || name.endsWith(".tar")
                                         || name.endsWith(".tgz")
                                         || name.endsWith(".dst")
                                         || name.endsWith(".jar")
-                                        || name.endsWith(".zip"));
+                                        || name.endsWith(".zip"))
+                .collect(Collectors.toList());
```

Comment:
The collect isn't necessary, we can use the `.count()` instead.

And the fieldName can be `archivedFileCount`, and return `archivedFileCount == shipFiles.size()`.